integrations/integration.php

```diff
@@ -1,66 +1,298 @@
+	 *   'site'          => array<string, mixed>,
+	 *   'network_sites' => array<string, array<string, mixed>>,
+	 * }
+	 *
+	 * @example
+	 * array(
+	 *  'client'        => array( 'status' => 'blocked' ),
+	 *  'site'          => array(
+	 *      'status' => 'enabled',
+	 *      'config'  => array(),
+	 *   ),
+	 *  'network_sites' => array (
+	 *      1 => array (
+	 *          'status' => 'disabled',
+	 *          'config'  => array(),
+	 *      ),
+	 *      2 => array (
+	 *          'status' => 'enabled',
+	 *          'config'  => array(),
+	 *      ),
+	 *  )
+	 * );
+	 */
+	private array $vip_config = [];
 
 	/**
-	 * A boolean indicating if this integration should be loaded. Defaults to false.
+	 * A boolean indicating if this integration is activated by customer.
 	 *
 	 * @var bool
 	 */
-	protected bool $is_active = false;
+	protected bool $is_active_by_customer = false;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $slug Slug of the integration.
+	 */
+	public function __construct( string $slug ) {
+		$this->slug = $slug;
+	}
 
 	/**
 	 * Activates this integration with an optional configuration value.
 	 *
-	 * @param array  $config An associative array of configuration values for the integration.
+	 * @param array $config An associative array of configuration values for the integration.
 	 *
 	 * @private
 	 */
 	public function activate( array $config = [] ): void {
-		$this->is_active = true;
-		$this->config    = $config;
+		$this->is_active_by_customer = true;
+		$this->customer_config       = $config;
 	}
 
 	/**
 	 * Returns true if this integration has been activated.
 	 *
+	 * @return bool
+	 *
 	 * @private
 	 */
 	public function is_active(): bool {
-		return $this->is_active;
+		if ( $this->is_active_by_customer ) {
+			return true;
+		}
+
+		return $this->is_active_by_vip();
+	}
+
+	/**
+	 * Returns true and passed available config if the integration is active by VIP.
+	 *
+	 * @return bool
+	 */
+	protected function is_active_by_vip(): bool {
+		$vip_config = $this->get_vip_config_from_file();
+		if ( ! is_array( $vip_config ) ) {
+			return false;
+		}
+
+		$this->vip_config = $vip_config;
+
+		// Return false if blocked on client.
+		if ( $this->get_value_from_vip_config( 'client', 'status' ) === Client_Integration_Status::BLOCKED ) {
+			return false;
+		}
+
+		$site_status = $this->get_value_from_vip_config( 'site', 'status' );
+
+		// Return false if blocked on site.
+		if ( Site_Integration_Status::BLOCKED === $site_status ) {
+			return false;
+		}
+
+		// If network site then look in network_sites config.
+		if ( is_multisite() ) {
+			$network_site_status = $this->get_value_from_vip_config( 'network_sites', 'status' );
+			
+			// Setup config and return true If enabled on network site.
+			if ( Site_Integration_Status::ENABLED === $network_site_status ) {
+				$have_config = $this->get_vip_config_of_current_network_site() !== '';
+
+				if ( $have_config ) {
+					$this->setup_config();
+				}
```

Comment:
Apologies, didn't get it can you please explain a bit more :)

Reply:
https://en.wikipedia.org/wiki/Side_effect_(computer_science)

Basically, the goal of the function is to give you a true or false value. However, the function does more work than that by changing the state of the object (this is the "side effect"), which is a bit unexpected. And because it's unexpected, that can lead to other unexpected behaviours and bugs.

Reply:
Got it. Thanks for explaning, will take care of this.

Reply:
Addressed in https://github.com/Automattic/vip-go-mu-plugins/pull/4660/commits/0f6f3967c49b9cd9e84a6432566b2a1d81a14417